### PR TITLE
docs: use new standard Open edX for documentation

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
**Description:**
This PR generates the new files structure for docs proposed in [this](https://docs.openedx.org/en/latest/developers/how-tos/add-sphinx-docs-to-a-repo.html) how-to by the community members. This is how it looks rendered with a few modifications:
image

**ISSUE:** 
https://github.com/openedx/openedx-events/issues/96

**Dependencies:** 
None for now.

**Merge deadline:** 
ASAP.

**Installation instructions:** 
Just documentation

**Testing instructions:**
Just documentation

**Reviewers:**
- [ ] @felipemontoya 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:** 
None for now.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi!

As a core committer in this repo, you can merge this once the pull request is approved per the [core committer reviewer requirements](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Open+edX+Core+Committers#Reviews-and-Ownership) and according to the agreement with your edX Champion.

File: requirements/constraints.txt
Comment (mariajgrimaldi) [inline requirements/constraints.txt:13]: FYI @feanil: when I generated the docs folder, I used just the requirements specified in the [how-to](https://docs.openedx.org/en/latest/developers/how-tos/add-sphinx-docs-to-a-repo.html?highlight=sphinx#steps). I didn't realize that libraries check rst formats [using doc8](https://github.com/openedx/openedx-events/blob/main/tox.ini#L56) which latest version has some compatibility issues with `sphinx-book-theme==0.3.3` so I had to include this constraint in my doc requirements.

Comment (feanil) [reply requirements/constraints.txt:13]: ok, thanks for the heads up.  They're pretty close to releasing sphinx-book-theme `0.4.0` so hopefully you won't need this for long.

**Comment by mphilbrick211** [comment]
@felipemontoya friendly follow-up on this :)

CC: @mariajgrimaldi

File: docs/conf.py
Comment (felipemontoya) [inline docs/conf.py:109]: nit: eol

File: requirements/doc.in
Comment (felipemontoya) [inline requirements/doc.in:1]: in the guide the file is called `docs` but it is called `doc` in edx-platform and some other repos I picked at random.

**Review by felipemontoya — APPROVED** [review]
This PR looks good to me. Since it took me like forever to review this probably the requirement are out of date now.

File: requirements/doc.in
Comment (mariajgrimaldi) [reply requirements/doc.in:1]: The Open edX cookiecutter generates a `doc.in` file, so that one is correct, at least in our context. We could open a PR to docs.openedx.org, changing it to the most common one

**Comment by openedx-webhooks** [comment]
@mariajgrimaldi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-events&entry.867055334=https://github.com/openedx/openedx-events/pull/169&entry.1484655318=https://github.com/mariajgrimaldi&entry.752974735=2023-01-18+15:39&entry.1917517419=2023-02-28+15:41&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**
